### PR TITLE
FE-2008 deprecate remaining text inputs

### PR DESCRIPTION
**Pull request by edleeks87**
# Description
Deprecates the remaining text input components (textarea, fieldset, number, decimal and grouped-character) components.
Moves to the `__deprecated__` directory, stories have been moved to `__deprecated__` section and components have been moved to relevant cypress build section

### CODE REVIEW

File: cypress/support/step-definitions/textbox-steps.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { label } from '../../locators';
 const TEXT_ALIGN = 'text-align';
 
 Then('Textbox placeholder is set to {string}', (text) => {
-  textbox().should('have.attr', 'placeholder', text);
+  // textbox().should('have.attr', 'placeholder', text);
```
Comment (kyanmichael) [inline cypress/support/step-definitions/textbox-steps.js]: Does this commented-out line need to remain here for future reference, or can it be deleted?

File: src/__deprecated__/components/decimal/decimal.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import I18nHelper from '../../utils/helpers/i18n';
-import Input from '../../utils/decorators/input';
-import InputLabel from '../../utils/decorators/input-label';
-import InputValidation from '../../utils/decorators/input-validation';
-import { validProps } from '../../utils/ether';
-import tagComponent from '../../utils/helpers/tags';
-import Logger from '../../utils/logger';
+import I18nHelper from '../../../utils/helpers/i18n/i18n';
```
Comment (kyanmichael) [inline src/__deprecated__/components/decimal/decimal.js]: What's the reason for making this category of changes in this PR?

```diff
- import I18nHelper from    '../../utils/helpers/i18n';
+ import I18nHelper from '../../../utils/helpers/i18n/i18n';
                                                      ^^^^
                      Why is this extra `i18n` added here?
```

The `src/utils/helpers/i18n/` directory has a `package.json` file containing `{ "main": "./i18n.js" }`, which specifies that doing an `import` of `utils/helpers/i18n` should load the `i18n.js` file.

So I don't think it's necessary to include that extra `/i18n` in the import path. But I could be wrong!

File: cypress/support/step-definitions/textbox-steps.js (hunk 1 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/textbox-steps.js]: Yes, it should be deleted :) My fault

File: src/__deprecated__/components/decimal/decimal.js (hunk 2 above)
Comment (edleeks87) [reply src/__deprecated__/components/decimal/decimal.js]: Not sure I just moved the files across so I guess this is a hangover form previous work, will update

Comment (edleeks87) [reply src/__deprecated__/components/decimal/decimal.js]: I've trimmed the paths for all the imports as they all seem to not make use of the `package.json` files in their parent directories

Comment (edleeks87) [reply src/__deprecated__/components/decimal/decimal.js]: https://github.com/Sage/carbon/pull/2310/commits/46a45e27dea3236888d3af10d28dd8d2a8545e72

File: cypress/support/step-definitions/textbox-steps.js (hunk 1 above)
Comment (kyanmichael) [reply cypress/support/step-definitions/textbox-steps.js]: No worries! Thanks for removing it in fb3c1b2 :thumbsup:

File: src/__deprecated__/components/decimal/decimal.js (hunk 2 above)
Comment (kyanmichael) [reply src/__deprecated__/components/decimal/decimal.js]: Many thanks :thumbsup:

There are a few more files which need their import paths trimmed:

**`src/__deprecated__/components/fieldset/__definition__.js`**
```diff
- import OptionsHelper from    '../../utils/helpers/options-helper';
+ import OptionsHelper from '../../../utils/helpers/options-helper/options-helper';
                                                                  ^^^^^^^^^^^^^^^

- import Definition from  './../../../demo/utils/definition';
+ import Definition from '../../../../demo/utils/definition/definition';
                                                           ^^^^^^^^^^^
```

**`src/__deprecated__/components/grouped-character/grouped-character.js`**
```diff
- import tagComponent from    '../../utils/helpers/tags';
+ import tagComponent from '../../../utils/helpers/tags/tags';
                                                       ^^^^^
```

**`src/__deprecated__/components/grouped-character/grouped-character.stories.js`**
```diff
- import OptionsHelper from    '../../utils/helpers/options-helper';
+ import OptionsHelper from '../../../utils/helpers/options-helper/options-helper';
                                                                  ^^^^^^^^^^^^^^^

- import getDocGenInfo from    '../../utils/helpers/docgen-info';
+ import getDocGenInfo from '../../../utils/helpers/docgen-info/docgen-info';
                                                               ^^^^^^^^^^^^
```

Comment (edleeks87) [reply src/__deprecated__/components/decimal/decimal.js]: Good spot, adjusted the paths

Comment (edleeks87) [reply src/__deprecated__/components/decimal/decimal.js]: https://github.com/Sage/carbon/pull/2310/commits/7d299aea100ad2e5ce0acf72d0b8c0af4ecc95f2

Comment (kyanmichael) [reply src/__deprecated__/components/decimal/decimal.js]: Perfect, many thanks :white_check_mark:

**Pull request merged**